### PR TITLE
adds conversations to the CLI

### PR DESCRIPTION
**Pull request by ryanfkeepers**
adds conversations support to groups backup commands

---

#### Does this PR need a docs update or release note?

- [x] :no_entry: No

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #4536

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by meain — APPROVED** [review]

File: src/cli/backup/groups_e2e_test.go [hunk 1]
```diff
@@ -293,19 +296,27 @@ func (suite *PreparedBackupGroupsE2ESuite) SetupSuite() {
...
 		)
 
 		switch set {
-		case channelMessages:
+		case path.ChannelMessagesCategory:
 			scopes = selTD.GroupsBackupChannelScope(sel)
 
-		case libraries:
+		case path.ConversationPostsCategory:
+			scopes = selTD.GroupsBackupChannelScope(sel)
```
Comment (pandeyabs) [inline src/cli/backup/groups_e2e_test.go:317]: Do we need a scope for conversations?

File: src/cli/utils/groups.go [hunk 2]
```diff
@@ -172,11 +179,13 @@ func AddGroupsFilter(
...
-		llf, lli    = len(opts.ListFolder), len(opts.ListItem)
-		lpf, lpi    = len(opts.PageFolder), len(opts.Page)
-		lg, lch, lm = len(opts.Groups), len(opts.Channels), len(opts.Messages)
+		groups     = opts.Groups
+		lfp, lfn   = len(opts.FolderPath), len(opts.FileName)
+		llf, lli   = len(opts.ListFolder), len(opts.ListItem)
+		lpf, lpi   = len(opts.PageFolder), len(opts.Page)
+		lg         = len(opts.Groups)
+		lch, lm    = len(opts.Channels), len(opts.Messages)
+		lcon, lpst = len(opts.Conversations), len(opts.Posts)
```
Comment (pandeyabs) [inline src/cli/utils/groups.go:188]: These acronyms are hard to follow through this func. This is one of the cases where I think we should just use `len(opts.Foo)` wherever they are being used. Otherwise it's easy to miss a letter and cause bugs. 

Also, the test (`TestIncludeGroupsRestoreDataSelectors`) only checks for include length. We should improve it to match against expected selectors.

**Review by pandeyabs — APPROVED** [review]
Looks good. Left a comment about var naming in one place and test improvements.

File: src/cli/backup/groups_e2e_test.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/cli/backup/groups_e2e_test.go:317]: Ooh, good catch!

File: src/cli/utils/groups.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/cli/utils/groups.go:188]: Yeah, they're getting out of hand, aren't they?  I'll clean that up.
